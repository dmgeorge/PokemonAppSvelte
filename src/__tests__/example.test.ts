import { tick } from 'svelte'
import Pokemon from '../routes/index.svelte'
import { describe, expect, it} from "vitest";

let host: HTMLElement

afterEach(() => {
  host.remove()
})

test('mount component', async () => {
  host = document.createElement('div')
  host.setAttribute('id', 'host')
  document.body.appendChild(host)
  const instance = new Pokemon({ target: host, props: { count: 4 } })
  expect(instance).toBeTruthy()
  expect(host.innerHTML).toContain('text-4xl text')
  expect(host.innerHTML).toMatchSnapshot()
  // const btn = host.getElementsByTagName('button')[0]
  // btn.click() // or btn.dispatchEvent(new window.Event('click', { bubbles: true }))
  // await tick()
  // expect(host.innerHTML).toContain('4 x 3 = 12')
  // btn.click()
  // await tick()
  // expect(host.innerHTML).toContain('4 x 4 = 16')
})

/*
//TODO improvements
    - generic way to create svelte components from import ( helper utility or library )
    - generic api for interacting with components
    - alternatives to expect with innerHTML
 */