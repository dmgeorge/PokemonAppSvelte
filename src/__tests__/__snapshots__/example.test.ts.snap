// Vitest Snapshot v1

exports[`mount component 1`] = `" <h1 class=\\"text-4xl text-center my-8 uppercase\\">Svelte kit Pokedex</h1> <input class=\\"w-full rounded-md text-lg p-4 border-2 border-gray-200\\" type=\\"text\\" placeholder=\\"Search Pokemon\\"> <div class=\\"py-4 grid gap-4 md:grid-cols-2 grid-cols-1\\"></div>"`;
